### PR TITLE
DEV-2285: Improve eyes look at

### PR DESCRIPTION
**Pull request by luiscuenca**
https://highfidelity.atlassian.net/browse/DEV-2285
This PR fixes the avatar look at algorithm and introduces some realistic blinking effect when the look at target changes.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Can someone from High Fidelity approve this PR?

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21823/HighFidelity-Beta-PR16317-fc01825.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21823/HighFidelity-Beta-Interface-PR16317-fc01825.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21823/HighFidelity-Beta-PR16317-fc01825.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21823/HighFidelity-Quest-Beta-PR16317-fc01825.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/21823/HighFidelity-Beta-PR16317-fc01825.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/21823/HighFidelity-Beta-Interface-PR16317-fc01825.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21902/HighFidelity-Beta-PR16317-505813b.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21902/HighFidelity-Quest-Beta-PR16317-505813b.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21902/HighFidelity-Beta-PR16317-505813b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21902/HighFidelity-Beta-Interface-PR16317-505813b.dmg

**Comment by luiscuenca** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21922/HighFidelity-Beta-PR16317-505813b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/21922/HighFidelity-Beta-PR16317-505813b.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/21922/HighFidelity-Beta-Interface-PR16317-505813b.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21922/HighFidelity-Beta-PR16317-505813b.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21922/HighFidelity-Quest-Beta-PR16317-505813b.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21974/HighFidelity-Beta-PR16317-9f0e82e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21974/HighFidelity-Beta-Interface-PR16317-9f0e82e.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21974/HighFidelity-Beta-PR16317-9f0e82e.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21974/HighFidelity-Quest-Beta-PR16317-9f0e82e.apk)

**Comment by luiscuenca** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21989/HighFidelity-Beta-PR16317-9f0e82e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/21989/HighFidelity-Beta-Interface-PR16317-9f0e82e.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/21989/HighFidelity-Beta-PR16317-9f0e82e.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21989/HighFidelity-Beta-PR16317-9f0e82e.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/21989/HighFidelity-Quest-Beta-PR16317-9f0e82e.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22045/HighFidelity-Beta-PR16317-a4fcb2c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22045/HighFidelity-Beta-Interface-PR16317-a4fcb2c.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22045/HighFidelity-Beta-PR16317-a4fcb2c.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22045/HighFidelity-Quest-Beta-PR16317-a4fcb2c.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22045/HighFidelity-Beta-PR16317-a4fcb2c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22045/HighFidelity-Beta-Interface-PR16317-a4fcb2c.exe

File: libraries/avatars/src/HeadData.cpp [hunk 1]
```diff
@@ -233,3 +233,31 @@ void HeadData::setHasProceduralEyeMovement(bool hasProceduralEyeMovement) {
 void HeadData::setFaceTrackerConnected(bool value) {
     _isFaceTrackerConnected = value;
 }
+
+void HeadData::setLookAtPosition(const glm::vec3& lookAtPosition) {
```
Comment (hyperlogic) [inline libraries/avatars/src/HeadData.cpp]: we still need the _lookAtPositionChanged timestamp to be set in all cases.  This is necessary for the network protocol to know if this field has been updated.  The timestamp might also need to be set in HeadData::updateLookAt().

File: libraries/avatars/src/HeadData.cpp [hunk 2]
```diff
@@ -233,3 +233,31 @@ void HeadData::setHasProceduralEyeMovement(bool hasProceduralEyeMovement) {
...
+        _requestLookAtPosition = lookAtPosition;
+    } else {
+        if (_lookAtPosition != lookAtPosition) {
+            _lookAtPositionChanged = usecTimestampNow();
+        }
+        _lookAtPosition = lookAtPosition;
+    }
+}
+
+void HeadData::updateEyeLookAt() {
```
Comment (hyperlogic) [inline libraries/avatars/src/HeadData.cpp]: Why is this delayed set to _lookAtPosition necessary?

Comment (luiscuenca) [reply libraries/avatars/src/HeadData.cpp]: A eye target reposition triggers a blink and it takes effect right when the eyelid is closed.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22084/HighFidelity-Beta-PR16317-9432a37.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22084/HighFidelity-Beta-Interface-PR16317-9432a37.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22084/HighFidelity-Beta-PR16317-9432a37.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22084/HighFidelity-Quest-Beta-PR16317-9432a37.apk)

**Comment by hyperlogic** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22092/HighFidelity-Beta-PR16317-9432a37.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22092/HighFidelity-Quest-Beta-PR16317-9432a37.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22092/HighFidelity-Beta-PR16317-9432a37.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22092/HighFidelity-Beta-Interface-PR16317-9432a37.dmg

**Comment by luiscuenca** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22094/HighFidelity-Beta-PR16317-9432a37.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22094/HighFidelity-Quest-Beta-PR16317-9432a37.apk)

**Comment by hyperlogic** [comment]
Windows build keeps failing for an unknown reason.  Perahps due to the dll fixup failing to find dependencies?

**Comment by hyperlogic** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22097/HighFidelity-Beta-PR16317-9432a37.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22097/HighFidelity-Quest-Beta-PR16317-9432a37.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22097/HighFidelity-Beta-PR16317-9432a37.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22097/HighFidelity-Beta-Interface-PR16317-9432a37.dmg

**Comment by hyperlogic** [comment]
windows build failed, compiler is out of heap space errors.

    c:\jenkins\workspace\pr-build\label\windows\interface\src\ui\keyboard.cpp(993): fatal error C1060: compiler is out of heap space [C:\jenkins\workspace\pr-build\label\windows\build\interface\interface.vcxproj]

build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22099/HighFidelity-Beta-PR16317-9432a37.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/22099/HighFidelity-Beta-Interface-PR16317-9432a37.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22099/HighFidelity-Beta-PR16317-9432a37.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22099/HighFidelity-Quest-Beta-PR16317-9432a37.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22099/HighFidelity-Beta-PR16317-9432a37.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22099/HighFidelity-Beta-Interface-PR16317-9432a37.exe

**Pull request merged**